waagent

```diff
@@ -2551,6 +2564,31 @@ class FreeBSDDistro(AbstractDistro):
         SetFileContents(location+"/ovf-env.xml", ovfxml)
         return retcode,out
 
+    def mkfile(self,filename,nbytes):
+        """
+        Create a non-sparse file of that size. Replace existing file. BSD-compatible dd version.
+        
+        For sizes < 64M, a single-pass dd is used. Otherwise use two-pass dd.
+        """
+        max_blocksize = 64 * 1024 ** 2
+        dd_cmd = "dd if=/dev/zero bs={bs} count={n} conv=notrunc of={fn}"
+        
+        if not type(nbytes) is int:
+           nbytes = int(nbytes)
+        
+        if os.path.isfile(filename):
+            os.remove(filename)
+        
+        fn_sh = ShellEscape(filename)
+        
+        if nbytes < max_blocksize:
+            exitcode = Run(dd_cmd.format(bs=bytes, fn=fn_sh, n=1))
+        else:
+            exitcode = Run(dd_cmd.format(bs=max_blocksize, fn=filename, n=bytes/max_blocksize))
+            exitcode <<= 8 # lshift so we have space for another exitcode
+            exitcode += Run(dd.cmd.format(bs=nbytes%max_blocksize, fn=fn_sh, n=1))
```

Comment:
First: typo. Should be Run(dd_cmd.format(....)) ; you have a dot between "dd" and "cmd". How did this work when you tested it?

Second: You only need to do this second running of the dd command if nbytes%max_blocksize is non-zero.